### PR TITLE
Create Models and Relationship

### PR DESCRIPTION
**Pull request by ghost**
## Description
Create Model and Relationship
## Reference Requirements
- Task ID: TI-1.2

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:
-Folder App/Models

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Models/Borrow.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    ];
+
+    /**
+     * Relationship belongsTo with User
+     *
+     * @return array
+     */
+    public function user()
+    {
+        return $this->belongsTo(User::class, 'category_id');
```
Comment (at-nale) [inline app/Models/Borrow.php]: change category_id  ->  user_id

File: app/Models/Book.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    public function posts()
+    {
+        return $this->hasMany(Post::class);
+    }
+
+    /**
+     * Relationship belongsToMany with User
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/Book.php]: return ko đúng
@return \Illuminate\Database\Eloquent\Relations\BelongsToMany

mấy cái dưới tương tự để ý return về `BelongsToMany` hay là gì.

Comment (at-haytran) [reply app/Models/Book.php]: ok anh, em sẽ sữa rồi push lên lại.

File: app/Models/Book.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        'publishing_year',
+        'language',
+        'quantity',
+        'count_rate'
+    ];
+
+    /**
+     * Relationship belongsTo with Category
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/Book.php]: @return \Illuminate\Database\Eloquent\Relations\BelongsTo

File: app/Models/Book.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    public function category()
+    {
+        return $this->belongsTo(Category::class, 'category_id');
+    }
+
+    /**
+     * Relationship hasMany with ImageBook
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/Book.php]: @return \Illuminate\Database\Eloquent\Relations\HasMany

File: app/Models/Book.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    public function imageBooks()
+    {
+        return $this->hasMany(ImageBook::class);
+    }
+
+    /**
+     * Relationship hasMany with Post
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/Book.php]: @return \Illuminate\Database\Eloquent\Relations\HasMany

File: app/Models/Borrow.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        'status',
+        'number_book',
+        'to_date',
+        'form_date'
+    ];
+
+    /**
+     * Relationship belongsTo with User
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/Borrow.php]: @return \Illuminate\Database\Eloquent\Relations\BelongsTo

File: app/Models/Borrow.php [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Relationship belongsToMany with Book
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function borrowDetails()
+    {
+        return $this->belongsToMany(Book::class, 'borrow_details', 'borrow_id', 'book_id')
+            ->withTimestamps();
```
Comment (at-huydq) [inline app/Models/Borrow.php]: no need

File: app/Models/Category.php [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    * @var array
+    */
+    protected $fillable = [
+        'name'
+    ];
+
+    /**
+     * Relationship hasMany with Book
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/Category.php]: @return \Illuminate\Database\Eloquent\Relations\HasMany

File: app/Models/ImageBook.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    */
+    protected $fillable = [
+        'book_id',
+        'image'
+    ];
+
+    /**
+     * Relationship belongsTo with Book
+     *
+     * @return array
```
Comment (at-huydq) [inline app/Models/ImageBook.php]: @return \Illuminate\Database\Eloquent\Relations\BelongsTo

Comment (at-huydq) [reply app/Models/ImageBook.php]: Mấy chỗ tương tự như vậy fix giúp anh luôn nhé

**Review by ghost — COMMENTED** [review]
Please check my cmt

File: app/Models/Book.php [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Relationship belongsToMany with User
+     *
+     * @return array
+     */
+    public function ratings()
+    {
+        return $this->belongsToMany(User::class, 'ratings', 'user_id', 'book_id')
+            ->withPivot('rate')->withTimestamps();
```
[the remark this answers is not in the document]
Comment (at-haytran) [reply app/Models/Book.php]: Cái này là quan hệ nhiều nhiều giữa User và Book mà anh.

**Comment by at-huydq** [comment]
LGTM.

**Pull request merged**